v2/yarpcfx/yarpcfx_test.go

```diff
@@ -63,6 +64,42 @@ func TestNewClientProvider(t *testing.T) {
 	})
 }
 
+func TestClientHasMiddleware(t *testing.T) {
+	var gotCallOder []string
+
+	var middleware = yarpc.UnaryOutboundTransportMiddlewareFunc(
+		func(ctx context.Context, _ *yarpc.Request, _ *yarpc.Buffer, o yarpc.UnaryOutbound) (*yarpc.Response, *yarpc.Buffer, error) {
+			gotCallOder = append(gotCallOder, "middleware")
+			return o.Call(ctx, nil, nil)
+		})
+
+	out := yarpctest.OutboundCallable(func(context.Context, *yarpc.Request, *yarpc.Buffer) (*yarpc.Response, *yarpc.Buffer, error) {
+		gotCallOder = append(gotCallOder, "outbound")
+		return nil, nil, nil
+	})
+	trans := &yarpctest.FakeTransport{}
+
+	client := yarpc.Client{
+		Name:  "client",
+		Unary: trans.NewOutbound(nil, yarpctest.OutboundCallOverride(out)),
+	}
+
+	result, err := NewClientProvider(ClientProviderParams{
+		UnaryOutboundTransportMiddleware: []yarpc.UnaryOutboundTransportMiddleware{middleware},
+		Clients: []yarpc.Client{client},
+	})
+
+	require.NoError(t, err)
+
+	client, ok := result.Provider.Client("client")
+	require.True(t, ok, "could not find client")
+
+	_, _, _ = client.Unary.Call(context.Background(), nil, nil)
+
+	wantCallOrder := []string{"middleware", "outbound"}
```

Comment:
This might be overkill, but we could add one more middleware (two total) to make sure that the application is happening in the order we expect.